src/app/components/translate-attributes/translate-attributes.component.ts

```diff
@@ -0,0 +1,341 @@
+            needsToBeUpdated = true;
+          }
+
+          if (needsToBeUpdated) {
+            this.multipleActionsPromises.push({
+              type: 'update',
+              id: attr.key,
+              data: attr,
+            });
+          }
+        } else {
+          console.log('attr.key', attr.key);
+          // New Items
+          if (!attr.key) {
+            throw new Error('Please ensure all Keys have a value.');
+          }
+          if (!attr['onesky-phrase-id']) {
+            throw new Error('Please ensure all Onesky Phase IDs have a value.');
+          }
+
+          this.multipleActionsPromises.push({
+            type: 'create',
+            id: attr.key,
+            data: attr,
+          });
+        }
+
+        // Ensure keys are unique
+        this.multipleActionsLocalAttributes.forEach((attribute) => {
+          if (attr.id === attribute.id) {
+            return;
+          }
+          if (attr.key === attribute.key) {
+            throw new Error(
+              '2 or more keys are the same. Please make sure all keys are unique.',
+            );
+          }
+        });
+      });
+      console.log('Update & Create checks - complete');
+
+      console.log('promises', this.multipleActionsPromises);
+      return {
+        successful: true,
+        message: 'Successful',
+      };
+    } catch (err) {
+      console.log('ERR', err);
+      return {
+        successful: false,
+        message: err.message,
+      };
+    }
+  }
+
+  multipleActions(): void {
+    this.multipleActionsLocalAttributes.clear();
+    this.multipleActionsRemoteAttributes.clear();
+    console.log('multipleActionsFunc');
+
+    this.attributeTranslationService
+      .getAttributes(this.resource.id)
+      .then(async (res) => {
+        try {
+          this.errorMessage = '';
+          this.multipleActionsError = false;
+          this.multipleActionsResults = [];
+          this.saving = true;
+          this.multipleActionsPromises = [];
+
+          const checkDifferences = await this.checkRemoteResourceForDifferences(
+            res,
+          );
+          if (!checkDifferences.successful) {
+            throw new Error(checkDifferences.message);
+          }
+
+          // Sort promises
+          await this.mulitipleActionSortPromises();
+
+          // Need to await for this
+          const sendPromises = new Promise(async (resolve, reject) => {
+            console.log('Start Promises');
+            for (const promise of this.multipleActionsPromises) {
+              console.log('promise.data', promise.data);
+              if (promise.type === 'delete') {
+                await this.mulitipleActionDelete(promise);
+              } else if (promise.type === 'update') {
+                await this.mulitipleActionUpdate(promise);
+              } else if (promise.type === 'create') {
+                await this.mulitipleActionCreate(promise);
+              }
+            }
```

Comment:
if I'm reading this right, it will fire off all updates in sequence because it `awaits` each promise in order.

because these updates are fast anyways and we have ordering concerns based on type, I'm fine with leaving this like this

Reply:
Yes, you are correct. I didn't want it to fire the promises in parallel, as we want to ensure all the `delete` promises are done before moving on to `update` and `create` promises.

Or are you referring to firing all the `delete` promises in parallel, then once complete, firing `update` promises and finally `create` promises?

Reply:
just was making sure I was reading it right and it was intentional.

It would be possible to fire the updates in 3 batches (all deletes, all updates, all creates), but I don't think it's worth reworking the logic for this.

No changes necessary here.